GenderPayGap.WebUI/Controllers/Admin/AdminRemoveUserController.cs

```diff
@@ -0,0 +1,130 @@
+        {
+            this.dataRepository = dataRepository;
+            RegistrationRepository = registrationRepository;
+            this.auditLogger = auditLogger;
+        }
+
+        [HttpGet("/remove-user-organisation")]
+        public IActionResult RemoveUserFromAnOrganisation(long orgId, long userId, bool fromViewUserPage)
+        {
+            UserOrganisation userOrg = dataRepository.GetAll<UserOrganisation>()
+                .FirstOrDefault(u => u.UserId == userId && u.OrganisationId == orgId);
+
+            ValidateUserOrganisationForRemoval(userOrg);
+
+            var viewModel = new AdminRemoveUserViewModel
+            {
+                OrganisationId = userOrg.OrganisationId,
+                OrganisationName = userOrg.Organisation.OrganisationName,
+                UserId = userOrg.UserId,
+                UserFullName = userOrg.User.Fullname,
+                FromViewUserPage = fromViewUserPage
+            };
+            return View("ConfirmRemoving", viewModel);
+        }
+
+        [PreventDuplicatePost]
+        [ValidateAntiForgeryToken]
+        [HttpPost("/remove-user-organisation")]
+        public async Task<IActionResult> RemoveUserFromAnOrganisation(AdminRemoveUserViewModel viewModel)
+        {
+            var user = dataRepository.Get<User>(viewModel.UserId);
+            var organisation = dataRepository.Get<Organisation>(viewModel.OrganisationId);
+            UserOrganisation userOrg = dataRepository.GetAll<UserOrganisation>()
+                .FirstOrDefault(u => u.UserId == viewModel.UserId && u.OrganisationId == viewModel.OrganisationId);
+
+            ValidateUserOrganisationForRemoval(userOrg);
+
+            viewModel.ParseAndValidateParameters(Request, m => m.Reason);
+
+            if (viewModel.HasAnyErrors())
+            {
+                // If there are any errors, return the user back to the same page to correct the mistakes
+                return View("ConfirmRemoving", viewModel);
+            }
+
+            await RegistrationRepository.RemoveRegistrationAsync(userOrg, user);
+
+            // Email user that has been unregistered
+            EmailSendingService.SendRemovedUserFromOrganisationEmail(
+                user.EmailAddress,
+                organisation.OrganisationName,
+                user.Fullname);
+
+            // Email the other users of the organisation
+            IEnumerable<string> emailAddressesForOrganisation = organisation.UserOrganisations.Select(uo => uo.User.EmailAddress);
+            foreach (string emailAddress in emailAddressesForOrganisation)
+            {
+                EmailSendingService.SendRemovedUserFromOrganisationEmail(
+                    emailAddress,
+                    organisation.OrganisationName,
+                    user.Fullname);
+            }
+
+            // Send the notification to GEO for each newly orphaned organisation
+            if (!user.EmailAddress.StartsWithI(Global.TestPrefix))

```

Comment:
We're moving away from using `Global.TestPrefix` (it was only ever used by some old load tests), so we can omit this `if`